kernel/src/main/java/org/kframework/compile/AddSortInjections.java

```diff
@@ -186,7 +186,7 @@ private K visitChildren(K term, Sort actualSort) {
                         List<Sort> polySorts = positions.stream()
                                 .map(p -> sort(kapp.items().get(p - 1), freshSortParam))
                                 .collect(Collectors.toList());
-                        expectedSort = lub(polySorts, null, kapp);
+                        expectedSort = lub(polySorts, null, kapp, mod);
```

Comment:
I don't get it: where is the extra argument `mod` being used?

Reply:
it's being used in upperBounds.

Reply:
and in the Collection overload of lub.